my-profile-local-contact-impl/src/main/java/edu/wisc/my/profile/local/service/LocalContactInformationServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package edu.wisc.my.profile.local.service;
+
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import edu.wisc.my.profile.local.dao.LocalContactInformationDao;
+import edu.wisc.my.profile.model.ContactInformation;
+import edu.wisc.my.profile.service.LocalContactInformationService;
+
+@Service
+public class LocalContactInformationServiceImpl implements LocalContactInformationService {
+  
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
+  
+  private LocalContactInformationDao dao;
+  
+  
+  private String adminGroup;
+  
+  @Value("${adminGroup}")
+  public void setAdminGroup(String group) {
+    adminGroup = group;
+  }
+  
+  @Autowired
+  public void setLocalContactInfoDao(LocalContactInformationDao lcio) {
+    this.dao = lcio;
+  }
+  
+
+  @Override
+  public ContactInformation getContactInfo(String netId) {
+    return dao.getData(netId);
+  }
+
+  @Override
+  public ContactInformation setContactInfo(String netId, ContactInformation contactInformation) throws Exception {
+    contactInformation.setLastModified(DateTime.now());
+    return dao.setData(netId, contactInformation);
+  }
+
+
+  
+  @Override
+  public ContactInformation getContactInfo(String username, String manifestGroups, String netId) {
+    if(StringUtils.isNotBlank(adminGroup) && manifestGroups.contains(adminGroup)) {
+      logger.info("User {} looked up user {}.",username, netId);
```

Comment:
Love the audit trail logging and agree this logging should be here and agree we could route this using logback configuration.

But I'm still going to advocate hard for the as-whom-are-we-asking being part of the API call to the backing Middleware APIs and for Middleware being responsible for keeping the audit trail log on this, since that's where that logging should be happening and that logging should happen regardless of how users access this API, whether through the portal or not.